cpp/src/parquet/level_conversion.cc

```diff
@@ -137,7 +137,9 @@ void DefLevelsToBitmap(const int16_t* def_levels, int64_t num_def_levels,
   // is deleted in a follow-up release.
   if (level_info.rep_level > 0) {
 #if defined(ARROW_HAVE_RUNTIME_BMI2)
-    if (CpuInfo::GetInstance()->HasEfficientBmi2()) {
+    const CpuInfo* ci = CpuInfo::GetInstance();
+    // BMI2 (pext, pdep) is only efficient on Intel X86 processors.
```

Comment:
I would rather keep something like the `HasEfficientBmi2` abstraction because this may become more complicated in the future: https://issues.apache.org/jira/browse/ARROW-10548

Reply:
Restored `HasEfficientBmi2`.